great_expectations_cloud/agent/agent.py

```diff
@@ -114,6 +116,16 @@ def run(self) -> None:
         self._listen()
         print("Connection to GX Cloud has been closed.")
 
+    # ZEL-505: A race condition can occur if two or more agents are started at the same time
+    #          due to the generation of passwords for rabbitMQ queues. This can be mitigated
+    #          by adding a delay and retrying the connection. Retrying with new credentials
```

Comment:
Can we support two valid sets of creds?

Reply:
the login happens only once and lasts the duration of the session. I.e. a connection is not terminated when the credentials change

Reply:
oh I see. the race condition window is narrower than I was imaging. Thanks for clarifying